idl-compiler.py

```diff
@@ -1353,6 +1353,15 @@ def param_view_type(t):
         return t.name + join_template_view(t.template_parameters, additional_types)
 
 
+def basic_type(t):
+    if isinstance(t, BasicType):
+        return to_view(t.name)
+    elif isinstance(t, TemplateType):
+        assert len(t.template_parameters) == 1
+        assert t.name != "boost::variant" and t.name != "std::variant"
+        return param_view_type(t.template_parameters[0])
```

Comment:
I don't understand why this is called basic_type if it returns something for BasicType || variant.

Reply:
It's called this way since for `std::vector<T>` it will return `T`.
I'm not sure what would be a better name.  How about `element_type`?

Reply:
Or `item_type`?

Reply:
element_type should work. But then it should insist on its input being a vector (or other range).

Reply:
`value_type`